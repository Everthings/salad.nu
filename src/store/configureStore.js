import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import rootReducer from "./reducer";
import api from "./middleware/api";

export default function () {
  return configureStore({
    reducer: rootReducer,
    middleware: [...getDefaultMiddleware(), api],
  });
}
