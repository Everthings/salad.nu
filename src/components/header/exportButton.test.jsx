import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "./../../store/configureStore";
import Theme from "./../themes/theme";
import ExportButton from "./exportButton";

describe("ExportButton", () => {
  it("renders correctly", () => {
    const store = configureStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Theme>
            <ExportButton />
          </Theme>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
