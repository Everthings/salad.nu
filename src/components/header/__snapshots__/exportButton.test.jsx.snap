// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportButton renders correctly 1`] = `
<div
  className="sc-AxjAm ilxYOT btn"
>
  <a
    data-testid="export-button"
    download="nu_schedule.csv"
    href="data:text/csv;charset=utf-8,ï»¿\\"Subject\\",\\"Start Date\\",\\"Start Time\\",\\"End Date\\",\\"End Time\\",\\"Location\\""
    onClick={[Function]}
    target="_self"
    title="Export Schedule to CSV (click About for more info)"
  >
    <div
      className="sc-AxirZ llpCmO"
    >
      Export
    </div>
  </a>
</div>
`;
